Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -94,6 +96,38 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
 
         #endregion
 
+        /// <inheritdoc/>
+        public Task<OIdCToken> GetOIdCTokenAsync(OIdCTokenOptions options, CancellationToken cancellationToken = default)
+        {
+            options.ThrowIfNull(nameof(options));
+            // If at some point some properties are added to OIdCToken that depend on the token having been fetched
+            // then initialize the token here.
+            TokenRefreshManager tokenRefreshManager = null;
+            tokenRefreshManager = new TokenRefreshManager(
+                ct => RefreshOIdCTokenAsync(tokenRefreshManager, options, ct), Clock, Logger);
+            return Task.FromResult(new OIdCToken(tokenRefreshManager));
+        }
+
+        private async Task<bool> RefreshOIdCTokenAsync(TokenRefreshManager caller, OIdCTokenOptions options, CancellationToken cancellationToken)
+        {
+            string uri = $"{GoogleAuthConsts.ComputeOidcTokenUrl}?audience={options.TargetAudience}";
+            if (options.TokenFormat == OIdCTokenFormat.Full || options.TokenFormat == OIdCTokenFormat.FullWithLicences)
+            {
+                uri = $"{uri}&format=FULL";
```

Comment:
just one note, the GCP metadta server will support just `&includeEmail=` flag which just includes the email info in the OIDC token.  THe `format=full` includes a lot of stuff advertizes internal info that shoudn't be there from the start. 
pls see internal ref:
b/147875742

that new query parameter `includeEmail=` doesn't exist yet so the only option is format=full...

@bshaffer - iwe should file issues for google-cloud-node, google-cloud-java to use the new metadata flag when its actually live

Reply:
I won't add includeEmail until it does exist.

OK with defaulting to full, and with lowercaising, I just saw it uppercased somewhere. TRUE should be lowercased as well for licenses I suppose. I'll just look at what node is doing and do the same.